components/ledger/pkg/api/routes/routes.go

```diff
@@ -37,6 +43,9 @@ func NewRouter(backend controllers.Backend, logger logging.Logger, healthControl
 		middleware.Recoverer,
 	)
 	router.Use(middlewares.Log())
+	if globalMetricsRegistry != nil {
```

Comment:
Maybe add the middleware before middleware.Recoverer? To avoid recover on panic and let the dedicated middleware do this job.